### ISSUE TITLE
[tool] Stop using pkg:usage

### ISSUE THREAD

**Issue by kevmoo** [flutter/flutter#150575]
(Rewritten by @andrewkolos)

- [X] Remove dependency on package:usage in the g3 `flutter` tool (or at least get the OK to delete it)
  - b/361811035
- [ ] Remove code dependent on package:usage in the `flutter` tool (packages/flutter_tools). Fix pre-existing bugs along the way.
  - https://github.com/flutter/flutter/pull/150667
  - https://github.com/flutter/flutter/pull/151359
  - https://github.com/flutter/flutter/pull/151517
  - https://github.com/flutter/flutter/pull/151518
  - https://github.com/flutter/flutter/pull/159705
  - https://github.com/flutter/flutter/pull/160060
  - https://github.com/flutter/flutter/pull/160369
  - https://github.com/flutter/flutter/pull/160663
  - https://github.com/flutter/flutter/pull/162279
  - https://github.com/flutter/flutter/pull/162354
  - https://github.com/flutter/flutter/pull/162468
  - https://github.com/flutter/flutter/pull/162550
  - https://github.com/flutter/flutter/pull/162627
  - https://github.com/flutter/flutter/pull/162646
  - https://github.com/flutter/flutter/pull/162648
  - https://github.com/flutter/flutter/pull/162658
- [ ] Remove package:usage dependency from packages/flutter_tools

## Original description (Jun 20)

The package is deprecated.
The analytics configuration is gone.

Zero reason to keep it, right?

**Comment by andrewkolos** [comment]
Probably, I don't mind making a quick attempt to remove this.

**Comment by mit-mit** [comment]
Yes, this can 100% be removed. I took down the old analytics end point.

**Comment by kevmoo** [comment]
@andrewkolos – you still cranking on this?

**Comment by andrewkolos** [comment]
> @andrewkolos – you still cranking on this?

Yes, but it's not one of my highest priorities. Here is the current status: https://github.com/flutter/flutter/pull/150667 represents the full change needed to resolve this issue. However—in my opinion—it's too large to be meaningfully reviewed, so I'm splitting it up into smaller PRs. The first of these is https://github.com/flutter/flutter/pull/151359, which is nearly ready for review.

**Comment by andrewkolos** [comment]
I've given up on this for now due to 1) g3's deep dependence on this code and 2) my existing backlog of P0s and P1s.
I've filed b/361811035 as a blocker for this, though—as far as I can tell—the fl@g team won't have the bandwidth to work on this either.

**Comment by mit-mit** [comment]
Thanks for the update Andrew! Are there any remaining non-g3 uses you know of?

**Comment by andrewkolos** [comment]
> Thanks for the update Andrew! Are there any remaining non-g3 uses you know of?

Other than the flutter tool? I wouldn't know. The tool itself still uses it (though it's effectively dead code). The PR I closed would have removed the remaining usages (most of it could be removed as-is right now—but this is something I plan to slowly chip away at as I find the time to do so).

**Comment by parlough** [comment]
Looks to be resolved by https://github.com/flutter/flutter/pull/184994 (landed as https://github.com/flutter/flutter/commit/461f43245df15d116966bcfeb8264d723df2c400) and some associated g3fix. Thanks for finishing this up Ben!

**Comment by kevmoo** [comment]
YES!!! Less than 2 years! 😁
